src/main/java/com/amuzil/omegasource/magus/radix/RadixTree.java

```diff
@@ -1,39 +1,44 @@
 package com.amuzil.omegasource.magus.radix;
 
 import com.amuzil.omegasource.magus.skill.forms.Form;
+import com.amuzil.omegasource.magus.skill.modifiers.api.Modifier;
+import com.amuzil.omegasource.magus.skill.modifiers.api.ModifierData;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.Entity;
+
+import java.util.List;
 
 public class RadixTree {
     private final Node root;
     private Node active;
+    private Form lastActivated = null;
+    private RadixPath path;
+    private Entity owner;
 
     public RadixTree(Node root) {
         this.root = root;
-//        this.branch = new Branch();
         this.active = root;
     }
 
-//    public <T> boolean registerLeaf(Class<Leaf<T>> type, Leaf<T> leaf) {
-//        return branch.registerLeaf(type, leaf);
-//    }
-
     public void burn() {
         if (active.terminateCondition() != null) {
             active.terminateCondition().unregister();
         }
 
         active = null;
-
-//        branch.burn();
     }
 
     public void start() {
-//        branch.reset(root);
         setActive(root);
+        path = new RadixPath();
     }
 
     private void setActive(Node node) {
         active = node;
 
+        if(active.getModifiers().size() > 0 && owner instanceof ServerPlayer player)
```

Comment:
Are you *sure* you want this player specific? I was under the impression we wanted to use this for all entities, and for npcs we just pass data in place of what the player would've done.

Reply:
this section should be player specific, cos its registering clientside listeners. All entities will still have their own ModifierData instances which can be passed data via their AI Goals instead of Listeners sending packets